File: backend/src/main/java/com/ll/hotel/domain/member/member/dto/response/BusinessResponse.java
package com.ll.hotel.domain.member.member.dto.response;

import com.ll.hotel.domain.member.member.entity.Business;
import com.ll.hotel.domain.member.member.type.BusinessApprovalStatus;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

public class BusinessResponse {
    public record ApprovalResult(
            Long businessId,
            String businessRegistrationNumber,
            LocalDate startDate,
            String ownerName,
            BusinessApprovalStatus approvalStatus
    ) {
        public static ApprovalResult of(Business business) {
            return new ApprovalResult(
                    business.getId(),
                    business.getBusinessRegistrationNumber(),
                    business.getStartDate(),
                    business.getOwnerName(),
                    business.getApprovalStatus()
            );
        }
    }

    public record Verification(
            List<Map<String, Object>> data
    ) {}
}

File: backend/src/main/java/com/ll/hotel/domain/member/member/dto/response/BusinessResponse.kt
package com.ll.hotel.domain.member.member.dto.response

import com.ll.hotel.domain.member.member.entity.Business
import com.ll.hotel.domain.member.member.type.BusinessApprovalStatus
import java.time.LocalDate

class BusinessResponse {
    data class ApprovalResult(
        val businessId: Long,
        val businessRegistrationNumber: String,
        val startDate: LocalDate,
        val ownerName: String,
        val approvalStatus: BusinessApprovalStatus
    ) {
        companion object {
            fun of(business: Business): ApprovalResult {
                return ApprovalResult(
                    business.id,
                    business.businessRegistrationNumber,
                    business.startDate,
                    business.ownerName,
                    business.approvalStatus
                )
            }
        }
    }

    data class Verification(
        val data: List<Map<String, Any>>
    )
}